corehq/apps/userreports/static/userreports/js/report_config.js

```diff
@@ -208,6 +214,36 @@ var reportBuilder = function () {
             return self.otherColumns().length > 0;
         });
 
+        self.serialize = function () {
+            return {
+                "report_title": self.reportTitle,
+                "report_description": "",  // TODO: self.reportDescription,
+                "report_type": self.reportType(),
+                "is_aggregation_enabled": self.isAggregationEnabled(),
+                "chart": self.selectedChart(),
+                "columns": _.map(self.selectedColumns(), function (c) { return c.serialize(); }),
+                "default_filters": [],  // TODO: self.defaultFilters,
+                "user_filters": [],  // TODO: self.userFilters,
+            };
+        };
+
+        self.saveButton = COMMCAREHQ.SaveButton.init({
+            unsavedMessage: "You have unsaved settings.",
+            save: function () {
+                self.saveButton.ajax({
+                    url: window.location.href,  // POST here; keep URL params
+                    type: "POST",
+                    data: JSON.stringify(self.serialize()),
+                    dataType: 'json',
+                });
+            }
+        });
+        self.saveButton.ui.appendTo($("#saveButtonHolder"));
```

Comment:
I hate that line. It couples the JS module too tightly to the HTML template. ... iirc I tried to do this in the template after reportConfig has been instantiated, and for some reason it didn't work, but I can't remember why. I'll try again.